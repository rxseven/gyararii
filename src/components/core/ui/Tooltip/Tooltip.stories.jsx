import React from 'react';
import { storiesOf } from '@storybook/react';

import Tooltip from './index';

const title = 'Components/Core/UI/Tooltip';

storiesOf(title, module)
  .add('top (default)', () => (
    <div>
      <p>
        Lorem Ipsum is simply dummy text of the printing and typesetting
        industry. Lorem Ipsum has been the industry’s standard dummy text ever
        since the 1500s.
      </p>
      <p>
        <a data-tip="Tooltip" href="/">
          Link Text
        </a>
      </p>
      <Tooltip />
    </div>
  ))
  .add('right', () => (
    <div>
      <p>
        Lorem Ipsum is simply dummy text of the printing and typesetting
        industry. Lorem Ipsum has been the industry’s standard dummy text ever
        since the 1500s.
      </p>
      <p>
        <a data-place="right" data-tip="Tooltip" href="/">
          Link Text
        </a>
      </p>
      <Tooltip />
    </div>
  ));
