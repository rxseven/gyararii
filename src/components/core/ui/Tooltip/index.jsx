import React from 'react';
import ReactTooltip from 'react-tooltip';

function Tooltip(props) {
  return <ReactTooltip {...props} effect="solid" place="top" />;
}

export default Tooltip;
