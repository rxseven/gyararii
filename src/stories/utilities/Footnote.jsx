import styled from 'styled-components';

const Footnote = styled.div`
  bottom: 2rem;
  font-size: 14px;
  position: fixed;
`;

export default Footnote;
