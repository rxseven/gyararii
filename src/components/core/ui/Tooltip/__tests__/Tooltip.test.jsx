import ReactTooltip from 'react-tooltip';
import { factory } from 'tests/utilities';
import Tooltip from '../index';

// Mock
jest.mock('react-tooltip');

// Arrange
const context = expect.any(Object);
const source = {};

// Setup
function setup(props) {
  return factory(Tooltip, source, props);
}

// Test suites
describe('<Tooltip />', () => {
  it('should render without crashing', () => {
    setup();
  });

  it('should render a tooltip correctly', () => {
    const expected = { arguments: { effect: 'solid', place: 'top' } };
    setup();

    expect(ReactTooltip).toHaveBeenCalledWith(expected.arguments, context);
  });
});
