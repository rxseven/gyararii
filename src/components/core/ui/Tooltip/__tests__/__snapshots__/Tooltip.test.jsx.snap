// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tooltip /> Snapshot tests should render correctly 1`] = `
<div
  className="__react_component_tooltip place-top type-dark "
  data-id="tooltip"
/>
`;
